src/app.rs

```diff
@@ -0,0 +1,49 @@
+use axum::{routing::get, Router};
+use sqlx::PgPool;
+use std::sync::Arc;
+use crate::contract_leaderboard::get_contracts;
+use crate::state::AppState;
+use crate::db::get_db_pool;
+
+pub async fn get_app() -> Router {
+    let db_pool = get_db_pool().await;
+    get_app_with_db_pool(db_pool)
+}
+
+fn get_app_with_db_pool(db_pool: PgPool) -> Router {
+    let shared_state = Arc::new(AppState { db_pool });
+    tracing_subscriber::fmt::init();
+    let app = Router::new()
+        .route("/contracts", get(get_contracts))
+        .with_state(shared_state);
+    app
+}
+
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use tower::ServiceExt;
+    use axum::{
+        body::Body,
+        http::Request,
+    };
+    use crate::contract_leaderboard::ContractEntry;
+
+    #[tokio::test]
+    async fn test_get_contracts_against_dev_db() {
```

Comment:
What do you think about running the tests just against the dev db. (Specified in env variable "DATABASE_URL").
Alternatively I could also set up test db connection pools and test against those. 
While being a bit more complex to set up, this would allow for more precise tests and reduce traffic / dependence on the remote DB connection.